packages/amplify_auth_cognito/lib/method_channel_auth_cognito.dart

```diff
@@ -31,8 +32,13 @@ class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito {
     try {
       return await _channel.invokeMethod('addPlugin');
     } on PlatformException catch (e) {
-      throw AmplifyException("Auth plugin has already been added, " +
-          "multiple plugins for Auth category are currently not supported.");
+      // Swallow PluginAddedHotRestartException to handle hot restart
+      if (e.code == AmplifyExceptionMessages.pluginHotRestartException) {
+        print("The auth plugin was detected in the underlying platform.");
+      } else {
+        throw AmplifyException("Auth plugin has already been added, " +
+            "multiple plugins for Auth category are currently not supported.");
+      }
```

Comment:
How do we know that it's the only other possibility?

Reply:
This was pre-existing code, but your point is correct.  I've replaced with:
```dart
 throw AmplifyException.fromMap(
            Map<String, String>.from(e.details));
```